src/backend/executor/nodeAgg.c

```diff
@@ -1118,6 +1118,15 @@ finalize_aggregate(AggState *aggstate,
 
 			result = FunctionCallInvoke(fcinfo);
 			*resultIsNull = fcinfo->isnull;
+			if (trunc_numeric_result_hook)
+			{
+			    result = trunc_numeric_result_hook(aggstate->ss.ps.plan, 
```

Comment:
Please make sure that either we have validated that we can traverse pointers here or put check like if(a) if (a->b) if (a->b->c) inside hook